### PR TITLE
helm: Allow setting issuer group for certificate in teleport-cluster

### PR DESCRIPTION
**Pull request by mantoine96**
Hi,

External issuers for cert-manager require setting a different issuer group. This is not currently supported by the Helm chart which doesn't allow for creating certificates with external issuers (such as AWS ACM PCA).

In this PR I've added the `group` argument to the `issuerRef` declaration, and I've added a default value in the `values.yaml` to `cert-manager.io` which is the default group for built-in issuers.

I just have one question: in order to actually issue a certificate from AWS ACM PCA I had to also add the `commonName` field. While this isn't necessary with most CAs it certainly doesn't hurt. Do you think it's worth adding to this PR?

### CODE REVIEW

**Comment by corkrean** [comment]
Hey @webvictim, could you take a look at this?

File: docs/pages/kubernetes-access/helm/reference.mdx [hunk 1]
```diff
@@ -546,6 +546,29 @@ scoped to a single namespace.
...
+  <TabItem label="values.yaml">
+  ```yaml
+  highAvailability:
+    certManager:
+      issuerGroup: cert-manager.io
+  ```
+  </TabItem>
+  <TabItem label="--set">
+  ```code
+  --set highAvailability.certManager.issuerKind=cert-manager.io
```
Comment (webvictim) [inline docs/pages/kubernetes-access/helm/reference.mdx:567]:
```suggestion
  --set highAvailability.certManager.issuerGroup=cert-manager.io
```

**Review by webvictim — DISMISSED** [review]
LGTM on the whole with the one change I suggested.

@thehunt33r Could you also please add a values file using `issuerGroup` under https://github.com/gravitational/teleport/tree/master/examples/chart/teleport-cluster/.lint so that the linter will check it?

File: docs/pages/kubernetes-access/helm/reference.mdx (hunk 1 above)
Comment (thehunt33r) [reply docs/pages/kubernetes-access/helm/reference.mdx]: @webvictim Thanks!

**Comment by thehunt33r** [comment]
@webvictim Thanks for the feedback. I have updated my branch. Sorry for the delay!

Do you think I should also add the `commonName` field (maybe make it optional so that the current behavior doesn't change)?

File: docs/pages/kubernetes-access/helm/reference.mdx [hunk 2]
```diff
@@ -546,6 +546,29 @@ scoped to a single namespace.
   </TabItem>
 </Tabs>
 
+### `highAvailability.certManager.issuerGroup`
+
+| Type | Default value | Can be used in `custom` mode? | `teleport.yaml` equivalent |
+| - | - | - | - |
+| `string` | `cert-manager.io` | ❌ | None |
+
+Sets the `Group` of `Issuer` to be used when issuing certificates with `cert-manager`. Defaults to `cert-manager.io` to use built-in issuers
```
Comment (webvictim) [inline docs/pages/kubernetes-access/helm/reference.mdx:555]:
```suggestion
Sets the `Group` of `Issuer` to be used when issuing certificates with `cert-manager`. Defaults to `cert-manager.io` to use built-in issuers.
```

**Review by webvictim — COMMENTED** [review]
@thehunt33r Looks good - one more tiny change. I need to copy and raise this PR myself so that CI will run it, so I'll do that later today.

Regarding `commonName`: if it's needed for AWS ACM PCA then it's probably worth adding, as I'd imagine that will be a common use case. I'd agree that it should be an optional field.

Many thanks for the changes!

**Comment by thehunt33r** [comment]
Thanks @webvictim 

I've added the optional `commonName` field in the certificate.yaml. Not super happy with the implementation but I'm open for your suggestions for more... elegant... ideas. Unfortunately I think it would be clunky to force users to instruct the `commonName` since it will almost always match the `clusterName`... but the boolean isn't great either.

**Review by webvictim — APPROVED** [review]
I think this is good now, the linter passed.

**Review by zmb3 — APPROVED** [review]

**Comment by webvictim** [comment]
Ping @r0mant @russjones @klizhentas, please approve.

**Review by russjones — APPROVED** [review]

**Pull request merged**